### PR TITLE
Toggle Cheat System Upgrade

### PR DESCRIPTION
**Pull request by lewri**
(3rd time's the charm?)
**Describe what the proposed change does**
- Moves toggle cheats into their own set (inside a private function, to be called when required)
- Removes cheat processing code from Fax
- Adjusts announceCheat to accommodate adviser text.
- active_cheats in hospital is now a set in Cheats.
- New cheat from fax to toggle fatigue off and on. Also sets max speed for all staff
  - Code to test: 363749
- Some luadoc.

### CODE REVIEW

File: CorsixTH/Lua/hospital.lua [hunk 1]
```diff
@@ -176,6 +176,9 @@ function Hospital:Hospital(world, avail_rooms, name)
   self.policies = {}
   self.discovered_diseases = {} -- a list
 
+  -- Create a cheats instance for each hospital. AIHospitals don't know how to cheat (we hope!)
+  self.hosp_cheats = Cheats(self)
```
Comment (TheCycoONE) [inline CorsixTH/Lua/hospital.lua:180]: Injection is better.

**Comment by TheCycoONE** [comment]
Looks good. Ignore my injection comment for now. That's a broader refactor - getting rid of all these accessing objects through other objects and passing them in on construction instead.

**Comment by ARGAMX** [comment]
I was looking for a cheat so that doctors don't get tired (no_rest_cheat) and didn't find it in the cheat menu.
But I knew that it was in the game from a player.

I also looked for a separate PR that added "don't get tired cheat", but I couldn't find it either. 🙂
In the end, it turned out that this PR adds this cheat. Although it's not obvious from the PR name.
I had to search for this commit through Git. 🙂

This cheat missing in `self.cheat_list` in `Cheats`.
Also I don't see this cheat in the cheat menu.
Why didn't we add this cheats to the cheat menu?

I also tried Ctrl+F on our main Wiki page and search for word "cheat", but no success.

For some reason, some cheats are in the menu, but others are not, and you might not even know they exist?

Screenshot 2026-07-19 at 09 47 11

**Comment by ARGAMX** [comment]
The `no_rest_cheat ` cheat is enabled, but the adviser tells the staff is getting tired...

Screenshot 2026-07-19 at 09 34 14

**Comment by lewri** [comment]
> I also tried Ctrl+F on our main Wiki page and search for word "cheat", but no success.
> 
> For some reason, some cheats are in the menu, but others are not, and you might not even know they exist?

Ah, they are fax cheats, intentionally obfuscated.
https://github.com/CorsixTH/CorsixTH/wiki/Cheats#fax-numbers-overview

<sup>Edit: Browser broke, link updated</sup>

**Pull request merged**